### PR TITLE
support for multilinestring layers

### PR DESCRIPTION
**Pull request by vermeulendivan**

### CODE REVIEW

File: stream_utilities.py [hunk 1]
```diff
@@ -1095,6 +1095,61 @@ def get_duplicate_points(layer, threshold):
...
 
 
+def check_if_multipart(input_layer):
+    """Checks if a layer stores multipart or singlepart features.
+
+    Checks if a layer is multipart or singlepart by checking the first
+    feature which does have a geometry
+
+    :param input_layer: A vector line layer.
+    :type input_layer: QGISVectorLayer
```
Comment (Samweli) [inline stream_utilities.py]:
```suggestion
    :type input_layer: QgsVectorLayer
```

File: stream_utilities.py [hunk 2]
```diff
@@ -1095,6 +1095,61 @@ def get_duplicate_points(layer, threshold):
...
+
+
+def convert_multipart_to_singlepart(input_layer):
+    """Converts a vector layer's multipart features to singlepart.
+
+    Converts each feature of the input_layer to a singlepart feature
+    and stores the singlepart feature in a temporary vector layer
+
+    :param input_layer: A vector line layer.
+    :type input_layer: QGISVectorLayer
```
Comment (Samweli) [inline stream_utilities.py]: same as above

File: stream_utilities.py [hunk 3]
```diff
@@ -1095,6 +1095,61 @@ def get_duplicate_points(layer, threshold):
...
+
+    :returns: input_layer converted to singlepart features.
+    :rtype: QGISVectorLayer
```
Comment (Samweli) [inline stream_utilities.py]: same as above

File: stream_utilities.py [hunk 4]
```diff
@@ -1095,6 +1095,61 @@ def get_duplicate_points(layer, threshold):
...
+    """
+    input_crs = input_layer.sourceCrs()
```
Comment (Samweli) [inline stream_utilities.py]:
```suggestion
    input_crs = input_layer.crs()
```

**Review by Samweli — CHANGES_REQUESTED** [review]
@vermeulendivan I have requested small modifications to these changes, also it will be nice if you could add tests for the new `check_if_multipart` and `convert_multipart_to_singlepart` functions.

**Comment by vermeulendivan** [comment]
@Samweli the two methods has been merged into one method now. made a change to the method which also now removes features with a null geometry. does this for both multilinestring and linestring files.

**Review by Samweli — APPROVED** [review]

**Pull request merged**